src/QsFmt/SPECIFICATION.md

```diff
@@ -0,0 +1,68 @@
+﻿# Q# Formatter Design Specification Document
+
+QsFmt is a source code formatter for Q#.
+
+## Where to get the formatter
+
+The `Microsoft.Quantum.QSharp.Formatter` NuGet package will be distributed with the rest of Q# on NuGet.org.
+As a dotnet tool, it is installed with the command `dotnet tool install Microsoft.Quantum.QSharp.Formatter`.
+
+## How to use the command line formatter
+
+After installing the tool the command `qsfmt` is used to run the formatter.
+There are two commands supported for the tool: 'format' and 'update'.
+The 'format' command runs the tool as a formatter for Q# code, running only those transformation rules that update whitespace and affect indentation. The underlying code will not be changed.
+The 'update' command allows the tool to be used as a means of updating old Q# code, replacing deprecated syntax with newer supported syntax. These transformation rules change the actual code given to them, and may occasionally fail.
+These two commands work separately and can't be run together. If the user wishes to both format and update their code, they would need to run the tool twice, first with 'update' and then with 'format'.
+Both of these commands should be idempotent, meaning that iterated uses of the commands will not continue to have effects on the inputs.
+
+The tool will initially only support the 'update' command, as the formatting functionality is still being worked on. The 'format' command will be unavailable until it is supported.
+
+## Input and Output
+Input files to the formatter will be the last argument(s) to the tool.
```

Comment:
Can stdin be passed as an input file, e.g. as `qsfmt format -`?

Reply:
From the [Fantomas documentation](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md) I see they have an option `--stdin` as well as an option `--stdout` for inputting and outputting to the standard streams. I hadn't seen much of an interest in this feature as of yet, the focus being on the formatting of files, so I didn't want to commit to having these options yet. They can always be added on afterwards if there is enough time and interest. Would you say that these options are pretty important to you? If they are important enough to people they can be added up front instead of later.

Reply:
Closing this thread since the immediate question seems to be answered - further conversations about the functionality in general can be part of the spec review later this months; we will merge a draft spec earlier, and update it based on that review as needed.